client/data/promote-post/use-promote-post-credit-balance-query.ts

```diff
@@ -9,7 +9,7 @@ const useCreditBalanceQuery = ( queryOptions = {} ) => {
 	return useQuery( {
 		queryKey: [ 'promote-post-credit-balance-siteid', selectedSiteId ],
 		queryFn: async () => {
-			if ( typeof selectedSiteId === 'number' ) {
+			if ( selectedSiteId ) {
```

Comment:
The main problem was that for atomic sites, for some reason, the siteId was a string.
It was numeric, but not a number.

I thought about parsing the number and check against NaN, but then I realized from the code, that we wanted to check for undefined here. (At least the error we give the user is that the Wpuserid is undefined).

In general, the user won't have much control over that ID, we read it from the store, so I thought it would be better if we trust that value, and just check for undefined here. If for some reason the value store is not correct, the balance response will give an error, and it will end up having the same result (us not showing the balance to the user).

Reply:
That makes a lot of sense. Let's leave it as-is.